client/tls_test.go

```diff
@@ -0,0 +1,164 @@
+package client_test
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/EventStore/EventStore-Client-Go/client"
+	"github.com/EventStore/EventStore-Client-Go/direction"
+	"github.com/EventStore/EventStore-Client-Go/position"
+)
+
+func TestTLSWithoutCertificateAndVerify(t *testing.T) {
+	container := GetEmptyDatabase()
+	defer container.Close()
+
+	config, err := client.ParseConnectionString(fmt.Sprintf("esdb://admin:changeit@%s?tls=true&tlsverifycert=false", container.Endpoint))
+	if err != nil {
+		t.Fatalf("Unexpected configuration error: %s", err.Error())
+	}
+
+	c, err := client.NewClient(config)
+	if err != nil {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+
+	err = c.Connect()
+	if err != nil {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+
+	numberOfEventsToRead := 1
+	numberOfEvents := uint64(numberOfEventsToRead)
+	_, err = c.ReadAllEvents(context.Background(), direction.Backwards, position.StartPosition, numberOfEvents, true)
+	require.NoError(t, err)
+}
+
+func TestTLSWithoutCertificate(t *testing.T) {
+	container := GetEmptyDatabase()
+	defer container.Close()
+
+	config, err := client.ParseConnectionString(fmt.Sprintf("esdb://admin:changeit@%s?tls=true&tlsverifycert=true", container.Endpoint))
+	if err != nil {
+		t.Fatalf("Unexpected configuration error: %s", err.Error())
+	}
+
+	c, err := client.NewClient(config)
+	if err != nil {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+
+	err = c.Connect()
+	if err != nil {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+
+	numberOfEventsToRead := 1
+	numberOfEvents := uint64(numberOfEventsToRead)
+	_, err = c.ReadAllEvents(context.Background(), direction.Backwards, position.StartPosition, numberOfEvents, true)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "certificate signed by unknown authority")
+}
+
+func TestTLSWithCertificate(t *testing.T) {
+	container := GetEmptyDatabase()
+	defer container.Close()
+
+	config, err := client.ParseConnectionString(fmt.Sprintf("esdb://admin:changeit@%s?tls=true&tlsverifycert=true", container.Endpoint))
```

Comment:
**SUGGESTION:** It might be worth checking also if `tls=true` and `tlsverifycert=true` are not provided if the defaults will be applied correctly.

Reply:
Added.